lib/http2/stream.c

```diff
@@ -135,15 +135,15 @@ static int send_data_pull(h2o_http2_conn_t *conn, h2o_http2_stream_t *stream)
     /* obtain content */
     cbuf.base = conn->_write.buf->bytes + conn->_write.buf->size + H2O_HTTP2_FRAME_HEADER_SIZE;
     cbuf.len = max_payload_size;
-    is_final = h2o_pull(&stream->req, stream->_pull_cb, &cbuf);
+    send_state = h2o_pull(&stream->req, stream->_pull_cb, &cbuf);
```

Comment:
I believe you would need to set `send_rst_stream_on_close` to true in case `h2o_pull` returns `H2O_SEND_STATE_ERROR`, otherwise no frame that closes the stream would be sent.

An alternative approach would be to pass the send_state as an argument to `encode_data_header_and_consume_window`, and encode the `RST_STREAM` frame within the function. I _think_ that would make the code little more concise.